build/scripts/AppExtensionsHelper.psm1

```diff
@@ -0,0 +1,264 @@
+function GetSourceCodeFromArtifact() {
+    param(
+        [string] $App,
+        [string] $TempFolder
+    )
+    $sourceArchive = Get-ChildItem -Path $TempFolder -Recurse -Filter "$App.Source.zip" -ErrorAction SilentlyContinue
+    $sourceCodeFolder = "$TempFolder/$($App -replace " ", "_")Source"
+
+    if (-not $sourceArchive) {
+        # Find out which version of the apps we need
+        if ($env:artifact) {
+            Write-Host "Found artifact: $($env:artifact)"
+            $artifact = $env:artifact
+        } else {
+            Write-Host "No artifact found. Using default artifact version."
+            Import-Module $PSScriptRoot\EnlistmentHelperFunctions.psm1
+            $artifact = Get-ConfigValue -ConfigType "AL-GO" -Key "artifact"
+        }
+        # Test that artifact is a url
+        if ($artifact -notmatch "^https?://") {
+            Write-Error "Artifact is not a valid URL: $artifact"
+            throw
+        }
+
+        $artifactVersion = $artifact -replace "/[^/]+$", "/w1"
+
+        # Download the artifact that contains the source code for those apps
+        Download-Artifacts -artifactUrl $artifactVersion -basePath $TempFolder -includePlatform | Out-Null
```

Comment:
Is there a way to cache this? Downloading it all the time seems unnecessary.

Alternatively, what about just switching to /w1 artifact altogether?

Reply:
> Is there a way to cache this? Downloading it all the time seems unnecessary.


You mean in the .githubcache or? We only download it once in the CLEAN mode build for add-ons. We don't download it at all in the other build modes 

> Alternatively, what about just switching to /w1 artifact altogether?

Ah yeah... You mean in the repo settings? Rather than using the base artifact?

Reply:
>You mean in the .githubcache or? We only download it once in the CLEAN mode build for add-ons. We don't download it at all in the other build modes

The cache that can be shared between dependencies: https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/caching-dependencies-to-speed-up-workflows

> Ah yeah... You mean in the repo settings? Rather than using the base artifact?
Indeed.

Reply:
> The cache that can be shared between dependencies: https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/caching-dependencies-to-speed-up-workflows

I can't call the cache action since the .yaml files are controlled by AL-Go 🤔 Our cache is also already full so I'm not sure we'll end up getting a lot out of caching base artifacts AND w1 artifacts 
https://github.com/microsoft/BCApps/actions/caches

We could switch to using the W1 artifact all over but that would also mean all the other projects would download the W1 artifact. Since the W1 artifact is 2,5 times bigger than the base artifact, it might end up hurting us more than it benefits us 🤷‍♂️

Reply:
Ok, then let's see how it goes as it is, and optimize if needed.